web/ui/react-app/src/pages/rules/RulePanel.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { FC, CSSProperties, Fragment } from 'react';
+import { Rule } from '../../types/types';
+import { Link } from '@reach/router';
+import { createExpressionLink, isPresent, replaceWith } from '../../utils';
+
+interface RulePanelProps {
+  rule: Rule;
+  styles?: CSSProperties;
+  tag?: keyof JSX.IntrinsicElements;
+}
+
+export const RulePanel: FC<RulePanelProps> = ({ rule, styles = {}, tag }) => {
+  const Tag = tag || Fragment;
+  const { name, query, labels, annotations, type, duration } = rule;
+  const style = { background: '#f5f5f5', ...styles };
+  const ruleName = replaceWith(name, `"`, `'`);
+  return (
+    <Tag {...(tag ? { style } : {})}>
+      <pre className="m-0" style={tag ? undefined : style}>
+        <code>
+          <div>
+            {/* Type of the rule is either alerting or recording. */}
+            {type.replace('ing', ': ')}
```

Comment:
This seems a bit too "clever" and relying on an accidental fact rather than a deeper domain invariant. I'd say let's not be afraid of being more explicit, even if longer:

```suggestion
            {type === 'record' ? 'recording' : 'alerting'}:{' '}
```

Reply:
I'm relying on the comments left in both rules structs regarding their type
https://github.com/prometheus/prometheus/blob/17d8c49919d4f26c422745b7a0a0ae5be4d63b72/web/api/v1/api.go#L908-L936

Reply:
Yeah, I'd still be more explicit though...

Reply:
Yep, thas fine. Will change it